cmake/HPX_AddCompileTest.cmake

```diff
@@ -18,14 +18,13 @@ function(add_hpx_compile_test category name)
 
   string(REGEX REPLACE "\\." "_" test_name "${category}.${name}")
 
-  add_hpx_library(
+  add_hpx_executable(
```

Comment:
Isn't this a compile-only test? Wouldn't `add_hpx_executable` attempt to link as well?

Reply:
Ok, saw your comment just now. I think this is fine as long as this does not cause linker errors.

Reply:
Good point, I hadn't thought of that. I think though that if we do have linker errors they will likely appear elsewhere as well.